### ISSUE TITLE
Deinterlacing "Bottom Frame First" footage in Double Framerate yields incorrect output

### ISSUE THREAD

**Issue by ray890** [obsproject/obs-studio#6061]
### Operating System Info

Windows 10

### OBS Studio Version

27.2.1

### OBS Studio Log URL

https://obsproject.com/logs/XatW6D21UZ_tunCz

### Expected Behavior

Sets of fields should be reconstructed in their appropriate frames, constructing the first set of fields in every odd frame, and the second set of fields in every even frame. This results in smooth video output with every frame in order, represented by the pattern of 1, 2, 3, 4, 5, 6, [...].

This would match the existing working OBS behavior when "Top Fields First" footage is deinterlaced in "Top Fields First" mode.

![image](https://user-images.githubusercontent.com/11812575/156057611-82b6cf45-fd2b-4802-8d0c-169c1146875e.png)

https://user-images.githubusercontent.com/11812575/156069727-a23ffc5b-bfc7-40cd-a7f8-4376223009c6.mp4

### Current Behavior

In the video output, top fields are delayed by 6 frames while bottom frames are constructed properly, resulting in a jerky video 
constructed out of order in the pattern of 1, 6, 3, 8, 5, 10, [...]

![image](https://user-images.githubusercontent.com/11812575/156062598-16237b52-3def-4648-9ae9-68ab37d6fc75.png)

https://user-images.githubusercontent.com/11812575/156070214-fe5a290e-9591-4f11-a307-64e6dd4c9d61.mp4

### Steps to Reproduce

1. Add an input source that contains video content that is interlaced using the "Bottom Field First" method
2. Enable the Deinterlacing using Blend 2x, Linear 2x or Yadif 2x
3. Set the Deinterlacing mode to "Bottom Field First"
4. Watch the video output

### Anything else we should know?

If a "Top Fields First" video were to be properly deinterlaced with "Bottom Fields First", and vice versa, then the second set of fields should be constructed one frame behind, leading to a mildly jerky video output. This is expected behavior, and not the behavior currently achieved with this bug.

![image](https://user-images.githubusercontent.com/11812575/156069904-ed4f6182-a612-4781-b3e5-fb2a68073be5.png)

https://user-images.githubusercontent.com/11812575/156070765-bb9e33e8-ffc1-4d74-8f77-e7c7602adaac.mp4

https://user-images.githubusercontent.com/11812575/156070243-0939d936-c4e2-4d9c-b8df-61edb971ef80.mp4

**Comment by faradey8951** [comment]
Agree! It really annoys me as well. I can't double the output framerate in OBS and it cause wrong output frame order

**Comment by rob-fi** [comment]
Chiming in to say this is still an issue. Running OBS 30.2.3 and attempting to use Yadif 2x deinterlacing for bottom-field first video (for example DV content). As detailed above and confirmed in my video below, this results in a big mess.

The Linear 2x does work but it doesn't look nearly as good as Yadif 2x.



https://github.com/user-attachments/assets/cca114fe-b2bb-420f-bb09-f5a80ea64ffd

**Comment by rob-fi** [comment]
I have since done some playing around.

In `C:\Program Files\obs-studio\data\libobs\deinterlace_base.effect` I modified the following line (58) to say field `1` rather than `0`:

```
float4 load_at(int2 texel, int x, int y, int field)
{
	if(field == 1)
		return load_at_image(texel, x, y);
	else
		return load_at_prev(texel, x, y);
}
```


It works! For BFF video that is. Of course, TFF deinterlacing was now broken. However, I did this following hack:

```
float4 load_at(int2 texel, int x, int y, int field)
{

	if(field != field_order)
		return load_at_image(texel, x, y);
	else
		return load_at_prev(texel, x, y);
}
```

And now it seems to work correctly for both TFF and BFF. I have absolutely no idea what the implications are of making this change, and I'll make the assumption that this is not the correct way to fix it until someone with more knowedge on the matter can chime in. But it's a workaround.
